plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -323,22 +334,104 @@ func (r *NodeAuthorizer) authorizeResourceSlice(nodeName string, attrs authorize
 		// is allowed by recording a graph edge.
 		return r.authorize(nodeName, sliceVertexType, attrs)
 	case "watch", "list", "deletecollection":
-		// Okay. The kubelet is trusted to use a filter for its own objects in watch and list.
-		// The NodeRestriction admission plugin (plugin/pkg/admission/noderestriction)
-		// ensures that the node is not deleting some ResourceSlice belonging to
-		// some other node.
-		//
-		// TODO (https://github.com/kubernetes/kubernetes/issues/125355):
-		// Once https://github.com/kubernetes/enhancements/pull/4600 is implemented,
-		// this code needs to be extended to verify that the node filter is indeed set.
-		// Then the admission check can be removed.
-		return authorizer.DecisionAllow, "", nil
+		if r.features.Enabled(features.AuthorizeNodeWithSelectors) {
+			// only allow a scoped fieldSelector
+			reqs, _ := attrs.GetFieldSelector()
+			for _, req := range reqs {
+				if req.Field == "nodeName" && req.Operator == selection.Equals && req.Value == nodeName {
```

Comment:
only the equal operator works here? do we validate the requirements to ensure its equals and return errors or errors when its not a supported operator?

Reply:
This is the internal in-memory authorization attributes representation, which only supports Equals/NotEquals today. 

in SubjectAccessReview structured API serialized form, we use In/NotIn, and we do validate when translating to/from authorization attributes.

Reply:
Do you mean https://github.com/kubernetes/kubernetes/pull/125571/files#diff-3e593550a88b0db529a9b865288964c112d96b6807e58fee301289206fb85398R145

Reply:
there are two paths to get authorization attributes:
1. SubjectAccessReview → authorization attributes translator - that's the helper method you linked to
2. normal `fieldSelector` query parameter parsing of actual list / watch API requests - that produces `hasTerm` instances which only support the Equals operator - https://github.com/kubernetes/kubernetes/blob/a8d354bf398a0f962179f4a8d1c4c8a70347190c/staging/src/k8s.io/apimachinery/pkg/fields/selector.go#L110-L116